File: 2491-divide-players-into-teams-of-equal-skill/2491-divide-players-into-teams-of-equal-skill.cpp
#define ll long long
class Solution {
public:
    long long dividePlayers(vector<int>& s) {
        sort(begin(s), end(s));
        unordered_set<ll> s1;
        ll ans = 0;
        int l = 0, r = s.size()-1;
        while(l<r){
            ll diff = s[l]+s[r];
            s1.insert(diff);
            ans += s[l]*s[r];
            l++,r--;
            if(s1.size()>1)     return -1;
        }
        return ans;
    }
};

File: 2491-divide-players-into-teams-of-equal-skill/2491-divide-players-into-teams-of-equal-skill.py
class Solution:
    def dividePlayers(self, skill: List[int]) -> int:
        skill.sort()
        l, r = 0, len(skill)-1
        req, ans = skill[l] + skill[r], 0
        while l<r:
            diff = skill[l] + skill[r]
            ans = ans + skill[l] * skill[r]
            l, r = l + 1, r - 1
            if req != diff:
                return -1
        return ans